Gems/O3DEMaterialEditor/Code/Include/O3DEMaterialEditor/O3DEMaterialEditorBus.h

```diff
@@ -0,0 +1,56 @@
+
+#pragma once
+
+#include <AzCore/EBus/Event.h>
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/std/function/function_template.h>
+#include <AzCore/std/string/string.h>
+
+class QWidget;
+
+namespace O3DEMaterialEditor
+{
+    using WidgetCreationFunc = AZStd::function<QWidget* (QWidget*)>;
+
+    class O3DEMaterialEditorRequests
+    {
+    public:
+        AZ_RTTI(O3DEMaterialEditorRequests, "{68bb1a2f-33b1-4906-adf3-c74c460400b1}");
+        virtual ~O3DEMaterialEditorRequests() = default;
+
+        //! Event signaled when it's ok to register views.
+        using NotifyRegisterViewsEvent = AZ::Event<>;
+
+        //! Connects a handler to the NotifyRegisterViewsEvent.
+        void ConnectNotifyRegisterViewsEventHandler(NotifyRegisterViewsEvent::Handler& handler)
+        {
+            handler.Connect(m_notifyRegisterViewsEvent);
+        }
+
+        //! Registers a view pane with the main editor.
+        virtual void RegisterViewPane(const AZStd::string& name, const WidgetCreationFunc& widgetCreationFunc) = 0;
+
+    protected:
+        NotifyRegisterViewsEvent m_notifyRegisterViewsEvent;
+    };
```

Comment:
Interface of O3DEMaterialEditor to do notifications and requests. For now I used the new AZ::Interface and AZ::Event classes for requests and notifications, instead of using buses. They are faster than buses.